apps/chat/src/contexts/SearchContext.tsx

```diff
@@ -148,112 +141,106 @@ export const SearchContextProvider = ({ children }: Props) => {
     // Save to history.
     setHistory(addHistoryItem({ query: value, filter, language }, history));
 
-    // First call - only search results - should come back quicky while we wait for summarization
     setIsSearching(true);
     setIsSummarizing(true);
     setSelectedSearchResultPosition(undefined);
 
-    let initialSearchResponse;
+    const startTime = Date.now();
+    let resultsWithSnippets;
 
     try {
-      const startTime = Date.now();
-      initialSearchResponse = await sendSearchRequest({
-        filter,
-        queryValue: value,
-        rerank: rerank.isEnabled,
-        rerankNumResults: rerank.numResults,
-        rerankerId: rerank.id,
-        rerankDiversityBias: rerank.diversityBias,
-        hybridNumWords: hybrid.numWords,
-        hybridLambdaLong: hybrid.lambdaLong,
-        hybridLambdaShort: hybrid.lambdaShort,
-        customerId: search.customerId!,
-        corpusId: search.corpusId!,
-        endpoint: search.endpoint!,
-        apiKey: search.apiKey!
-      });
-      const totalTime = Date.now() - startTime;
-
-      // If we send multiple requests in rapid succession, we only want to
-      // display the results of the most recent request.
-      if (searchId === searchCount) {
-        setIsSearching(false);
-        setSearchTime(totalTime);
-        setSearchResponse(initialSearchResponse);
-
-        if (initialSearchResponse.response.length > 0) {
-          setSearchError(undefined);
-        } else {
-          setSearchError({
-            message: "There weren't any results for your search."
-          });
+      const onStreamEvent = (event: ApiV2.StreamEvent) => {
+        switch (event.type) {
+          case "requestError":
+          case "genericError":
+            setSearchError({
+              message: "Error sending the query request"
+            });
+            break;
```

Comment:
This is my initial attempt at funneling the various errors into existing create-ui functionality, so we don't have to update any UIs or add features to each app.

Basically, anything that's generic or a request errors is shown as a search error. Previously, this error was set as the `catch` handler of the initial `sendSearchRequest` call.

Anything that's not a request or generic error will funnel into the existing summarization errors for now.

We can iterate on this as necessary, but this seemed to make the most sense to me.

Reply:
How about handling "unexpectedError" as a search error, too?